src/hotspot/share/gc/parallel/mutableSpace.cpp

```diff
@@ -113,14 +112,14 @@ void MutableSpace::initialize(MemRegion mr,
     }
     assert(mr.contains(head) && mr.contains(tail), "Sanity");
 
+    size_t page_size = UseLargePages ? alignment() : os::vm_page_size();
```

Comment:
I _think_ just using alignment should be sufficient here, and there's no need to conditionalize on UseLargePages.  But maybe alignment should be renamed to page_size?

Reply:
> I think just using alignment should be sufficient here

I did not want to change too much, but yes, I'll try that.

>  But maybe alignment should be renamed to page_size?

I wanted to keep the patch small, but I'll look into this too.

Reply:
I think it is actively wrong to be checking UseLargePages here.  That is done when the space is reserved, and the reservation succeeds or fails, with the actual page size determined by that.  Once we're here the decision of whether to use large pages is over with, and might have been in the negative, but without changing the flag.

Reply:
Changed. I agree.

Reply:
I was too quick changing the setter name from `alignment()` to `page_size()` - the subclass `MutableNUMASpace` already contains such a method, and that has been another reason to not do this. Please let me undo and look at this again.

Reply:
I pushed a change now that just fixes the issues, without renaming `alignment()`. I would like to keep this out of this change because `MutableNUMASpace` has its own `page_size()` already, which it may modify (and reallocate everything apparently)